stubs/Collection.stub

```diff
@@ -11,24 +11,27 @@ namespace Illuminate\Support;
 class Collection implements \ArrayAccess, Enumerable
 {
     /**
-     * @param  callable|null  $callback
-     * @param  mixed  $default
-     * @return TValue|null
+     * @template TDefault
+     * @param    callable|null  $callback
+     * @param    TDefault  $default
+     * @return   TValue|TDefault
      */
     public function first(callable $callback = null, $default = null){}
 
     /**
-     * @param  callable|null  $callback
-     * @param  mixed  $default
-     * @return TValue|null
+     * @template TDefault
+     * @param    callable|null  $callback
+     * @param    TDefault  $default
+     * @return   TValue|TDefault
      */
     public function last(callable $callback = null, $default = null){}
 
 
     /**
-     * @param  mixed  $key
-     * @param  mixed  $default
-     * @return TValue|null
+     * @template TDefault
+     * @param    mixed  $key
```

Comment:
I think this can be just `string` or `int` isn't it? In that case the type here can be `array-key` instead of `mixed`

Reply:
In my opinion, it should even be TKey. I didn't want to change it in this PR but if you want I can add it. I just don't know where to add tests for this

Reply:
I think you can test it like `assertType('*ERROR*', $collection->get(3.14));`

Reply:
Unfortunately, it doesn't work... Maybe I should add tests in `tests/Rules` and a file in `tests\Rules\Data` but I don't think this is the right place...